src/Compilers/VisualBasic/Portable/Semantics/AccessCheck.vb

```diff
@@ -448,36 +400,118 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
-        ' Determine if "derivedType" inherits from "baseType", ignoring constructed types, and dealing
-        ' only with original types.
-        Private Shared Function InheritsFromIgnoringConstruction(derivedType As TypeSymbol,
-                                                                 baseType As TypeSymbol,
-                                                                 basesBeingResolved As ConsList(Of Symbol),
-                                                                 <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo)) As Boolean
+        ''' <summary>
+        ''' Determine if "derivedType" inherits from or implements "baseType", ignoring constructed types, and dealing
+        ''' only with original types.
+        ''' </summary>
+        Private Shared Function InheritsFromOrImplementsIgnoringConstruction(derivedType As TypeSymbol,
+                                                                             baseType As TypeSymbol,
+                                                                             basesBeingResolved As BasesBeingResolved,
+                                                                             <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo)) As Boolean
             Debug.Assert(derivedType.IsDefinition)
             Debug.Assert(baseType.IsDefinition)
 
+            Dim interfacesLookedAt As PooledHashSet(Of NamedTypeSymbol) = Nothing
+            Dim baseInterfaces As ArrayBuilder(Of NamedTypeSymbol) = Nothing
+
+            Dim baseTypeIsInterface = baseType.IsInterfaceType()
+
+            If baseTypeIsInterface Then
+                interfacesLookedAt = PooledHashSet(Of NamedTypeSymbol).GetInstance()
+                baseInterfaces = ArrayBuilder(Of NamedTypeSymbol).GetInstance()
+            End If
+
             Dim current As TypeSymbol = derivedType
+            Dim result As Boolean = False
 
             While current IsNot Nothing
-                If current.Equals(baseType) Then
-                    Return True
+                If baseTypeIsInterface = current.IsInterfaceType() AndAlso current.Equals(baseType) Then
+                    result = True
+                    Exit While
+                End If
+
+                If baseTypeIsInterface Then
+                    AddBaseInterfaces(current, baseInterfaces, interfacesLookedAt, basesBeingResolved)
                 End If
 
-                If basesBeingResolved IsNot Nothing AndAlso basesBeingResolved.Contains(current) Then
+                If basesBeingResolved.InheritsBeingResolvedOpt?.Contains(current) Then
                     ' We can't obtain BaseType if we're currently resolving the base type of current.
                     current = Nothing ' can't go up the base type chain.
                 Else
+                    Select Case current.TypeKind
+                        Case TypeKind.Interface
+                            current = Nothing
```

Comment:
>current = Nothing [](start = 28, length = 17)

I didn't understand why we stop here. The interface could inherit other interfaces, one of which might be `baseType`.